File: 05-select-from-a-dropdown/java/src/test/java/Dropdown.java
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.ArrayList;
import java.util.List;

import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.MatcherAssert.assertThat;

public class Dropdown {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }

    @Test
    public void dropdownTest() {
        driver.get("http://the-internet.herokuapp.com/dropdown");
        WebElement dropdownList = driver.findElement(By.id("dropdown"));
        List<WebElement> options = dropdownList.findElements(By.tagName("option"));
        for (int i = 0; i < options.size(); i++) {
            if (options.get(i).getText().equals("Option 1")) {
                options.get(i).click();
            }
        }
        String selectedOption = "";
        for (int i = 0; i < options.size(); i++) {
            if (options.get(i).isSelected()) {
                selectedOption = options.get(i).getText();
            }
        }
        assertThat(selectedOption, is("Option 1"));
    }

    @Test
    public void dropdownTestRedux() {
        driver.get("http://the-internet.herokuapp.com/dropdown");
        Select selectList = new Select(driver.findElement(By.id("dropdown")));
        selectList.selectByVisibleText("Option 1");
        // You could also use select.selectByValue("1");
        assertThat(selectList.getFirstSelectedOption().getText(), is(equalTo("Option 1")));
    }

}

File: 05-select-from-a-dropdown/kotlin/src/test/kotlin/Dropdown.kt
import org.junit.Test
import org.openqa.selenium.By
import org.openqa.selenium.firefox.FirefoxDriver
import org.openqa.selenium.support.ui.Select
import kotlin.test.assertEquals
import kotlin.test.AfterTest

class Dropdown {
    private val testUrl = "http://the-internet.herokuapp.com/dropdown"

    private val driver by lazy {
        FirefoxDriver()
    }

    @AfterTest
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
    }

    @Test
    fun dropdownTest() {
        driver.get(testUrl)
        val dropdownList = driver.findElement(By.id("dropdown"))
        val options = dropdownList.findElements(By.tagName("option"))
        for (i in options.indices) {
            if (options[i].text == "Option 1") {
                options[i].click()
            }
        }
        for (i in options.indices) {
            if (options[i].isSelected) {
                assertEquals(
                        "Option 1",
                        options[i].text
                )
            }
        }

    }

    @Test
    fun dropdownTestRedux() {
        driver.get(testUrl)
        val selectList = Select(driver.findElement(By.id("dropdown")))
        selectList.selectByVisibleText("Option 1")
        // You could also use select.selectByValue("1");
        assertEquals(
                "Option 1",
                selectList.firstSelectedOption.text
        )
    }
}